src/tools/interop/idt/capture/shell_utils.py

```diff
@@ -0,0 +1,81 @@
+#
+#    Copyright (c) 2023 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+import shlex
+import subprocess
+
+from mobly.utils import stop_standing_subprocess
+
+
+class BashRunner:
+    """
+    Uses subprocess to execute bash commands
+    Intended to be instantiated and then only interacted with through instance methods
+    """
+
+    def __init__(self, command: str, sync: bool = False,
+                 capture_output: bool = False) -> None:
+        """
+        Run a bash command as a sub process
+        :param command: Command to run
+        :param sync: If True, wait for command to terminate
+        :param capture_output: Only applies to sync; if True, store stdout and stderr
+        """
+        self.command: str = command
+        self.sync = sync
+        self.capture_output = capture_output
+
+        self.args: list[str] = []
+        self._init_args()
+        self.proc: subprocess.Popen[bytes] | subprocess.CompletedProcess[bytes] | None = None
+
+        if self.sync:
+            self.proc = subprocess.run(
+                self.args, capture_output=capture_output)
+        else:
+            self.proc = subprocess.Popen(self.args)
+
```

Comment:
You could make the `proc` const.


```suggestion
        self.proc: subprocess.Popen[bytes] | subprocess.CompletedProcess[bytes] =  \
            subprocess.run(self.args, capture_output=capture_output) if self.sync  \
            else subprocess.Popen(self.args)
```

Reply:
Thanks, have taken care of this now!